### PR TITLE
Add Haiku build support

### PR DESCRIPTION
**Pull request by kenmays**
Adds OS build support for Haiku (https://www.haiku-os.org/).

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -323,7 +323,7 @@ fi
 
 
 dnl detect a usable system regular expression library
-AC_CHECK_LIB(c, regcomp, LIBREGEX="-lc",
+AC_CHECK_LIB(root, regcomp, LIBREGEX="-lroot",
```
Comment (jwmelto) [inline configure.ac:326]: I can't tell in context if this is a BeOS-specific portion or not, but I'm not sure it's a good idea to remove `-lc`; extending to check `-lroot` is fine.

File: configure.ac [hunk 2]
```diff
@@ -363,7 +363,7 @@ AC_SUBST(LIBRT)
 AC_CHECK_LIB(m, sqrtf)
 
 # for BeOS - old network stack don't have those libs ( move it in the case switch ?)
-AC_CHECK_LIB([socket], [socket])
+AC_CHECK_LIB([network], [socket])
```
Comment (jwmelto) [inline configure.ac:366]: same here; I'm not at all literate in configure. Should this be an additional test instead of a change?

**Review by jwmelto — COMMENTED** [review]
Wow. I remember being in the network code and seeing the BEOS defines and wondering if anyone would ever take that path. 

It's really cool to have you taking this on.

File: configure.ac (hunk 1 above)
Comment (kenmays) [reply configure.ac]: Fixed.

File: configure.ac (hunk 2 above)
Comment (kenmays) [reply configure.ac:367]: Fixed. Retest build. Works fine.

**Review by jwmelto — APPROVED** [review]
This looks fine to me

**Pull request closed without merging**